ipcore/dataplane_config.py

```diff
@@ -1,222 +1,51 @@
-                    }
-                }, 
-                {
-                    "name": "avalon_sink_error",
-                    "data_type": {
-                        "width": 2,
-                        "type": "ufix2",
-                        "signed": False,
-                        "fractional_bits": 0
-                    }
-                }
-            ]
-        }
-        legacy_dict["avalon_source"] = {
-            "signal": [
-                {
-                    "name": "avalon_source_valid",
-                    "data_type": {
-                        "width": 1,
-                        "type": "boolean",
-                        "signed": False,
-                        "fractional_bits": 0
-                    }
-                },
-                {
-                    "name": "avalon_source_data",
-                    "data_type": {
-                        "width": 32,
-                        "type": "sfix32_En28",
-                        "signed": model.system.audioOut.signed,
-                        "fractional_bits": 28
-                    }
-                },
-                {
-                    "name": "avalon_source_channel",
-                    "data_type": {
-                        "width": 2,
-                        "type": "ufix1",
-                        "signed": False,
-                        "fractional_bits": 0
-                    }
-                },
-                {
-                    "name": "avalon_source_error",
-                    "data_type": {
-                        "width": 2,
-                        "type": "ufix2",
-                        "signed": False,
-                        "fractional_bits": 0
-                    }
-                }
-            ]}
-        legacy_dict["avalon_memorymapped"] = {
-            "register" : generate_legacy_register(model.devices[0].registers)
-        }
-        return DataplaneConfig.parse_json_from_legacy(legacy_dict)
+        config.sink_max_channel = int(
+            ceil(log(model.system.audioIn.numberOfChannels, 2)))
+        config.source_max_channel = int(
+            ceil(log(model.system.audioOut.numberOfChannels, 2)))
+        return config
 
-def generate_legacy_register(new_registers):
-    old_registers = []
-    for reg in new_registers:
-        old_reg = {
-            "default_value": reg.defaultValue,
-                "name": reg.name,
-                "data_type": {
-                    "width": reg.dataType.wordLength,
-                    "type": "",
-                    "signed": reg.dataType.signed,
-                    "fractional_bits": reg.dataType.fractionLength
-                },
-            "reg_num": reg.registerNumber,
-            "max_value": 0,
-            "min_value": 0
-        }
-        old_registers.append(old_reg)
-    return old_registers
 
-def modelJsonDecoder(modelDict):
+def model_json_decoder(modelDict):
     return namedtuple('Model', modelDict.keys())(*modelDict.values())
```

Comment:
Just out of curiosity, how does this syntax work? It creates a named tuple with all of the dict keys, then somehow assigns values to each attribute in he named tuple?

Reply:
Yeah, so the `*` operator expands the list of values to be inputs to the the constructor. An easier to read example can be found in `autogen_hw_tcl.py` where I call `main(*parse_args())` to simplify passing arguments from `parse_args` to `main`.

Reply:
Right, that makes sense. I think I've used `*` for that purpose elsewhere, but I couldn't remember.